spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarConsumerFactory.java

```diff
@@ -16,26 +16,62 @@
 
 package org.springframework.pulsar.core;
 
+import java.util.Collection;
+import java.util.List;
 import java.util.Map;
 
-import org.apache.pulsar.client.api.BatchReceivePolicy;
 import org.apache.pulsar.client.api.Consumer;
 import org.apache.pulsar.client.api.PulsarClientException;
 import org.apache.pulsar.client.api.Schema;
 
+import org.springframework.lang.Nullable;
+
 /**
  * Pulsar consumer factory interface.
  *
  * @param <T> payload type for the consumer.
  * @author Soby Chacko
+ * @author Christophe Bornet
  */
 public interface PulsarConsumerFactory<T> {
 
-	Consumer<T> createConsumer(Schema<T> schema, Map<String, Object> propertiesToOverride) throws PulsarClientException;
+	/**
+	 * Create a consumer.
+	 * @param schema the schema of the messages to be sent
+	 * @return the consumer
+	 * @throws PulsarClientException if any error occurs
+	 */
+	Consumer<T> createConsumer(Schema<T> schema) throws PulsarClientException;
+
+	/**
+	 * Create a consumer.
+	 * @param schema the schema of the messages to be sent
+	 * @param topics the topics the consumer will subscribe to builder
+	 * @return the consumer
+	 * @throws PulsarClientException if any error occurs
+	 */
+	Consumer<T> createConsumer(Schema<T> schema, Collection<String> topics) throws PulsarClientException;
 
-	Consumer<T> createConsumer(Schema<T> schema, BatchReceivePolicy batchReceivePolicy,
-			Map<String, Object> propertiesToOverride) throws PulsarClientException;
+	/**
+	 * Create a consumer.
+	 * @param schema the schema of the messages to be sent
+	 * @param topics the topics the consumer will subscribe to or {@code null} to use the
+	 * default topics
+	 * @param properties the properties to attach to the consumer or {@code null} to use
+	 * the default properties
+	 * @param customizers the optional list of customizers to apply to the consumer
+	 * builder
+	 * @return the consumer
+	 * @throws PulsarClientException if any error occurs
+	 */
+	Consumer<T> createConsumer(Schema<T> schema, @Nullable Collection<String> topics,
```

Comment:
It may be helpful to enhance the comments here to include the fact (at a high level) that we set topics via the param and if you set them on the builder it will append to the default topics, not replace them.

Reply:
done